### PR TITLE
Add docs on Backup API type

### PR DESCRIPTION
**Pull request by ncdc**
Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

### CODE REVIEW

File: docs/api-types/backup.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    matchLabels:
+      app: ark
+      component: server
+  # Whether or not to snapshot volumes. This only applies to PersistentVolumes for Azure, GCE, and
+  # AWS. Valid values are true, false, and null/unset. If unset, Ark performs snapshots by default.
+  snapshotVolumes: null
+  # The amount of time before this backup is eligible for garbage collection.
+  ttl: 24h0m0s
+  # Actions to perform at different times during a backup. The only hook currently supported is
+  executing a command in a container in a pod using the pod exec API. Optional.
```
Comment (skriss) [inline docs/api-types/backup.md]: missing `#`

File: docs/api-types/backup.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  # PersistentVolumeClaim is included in the backup, its associated PersistentVolume (which is
+  # cluster-scoped) would also be backed up.
+  includeClusterResources: null
+  # Individual objects must match this label selector to be included in the backup. Optional.
+  labelSelector:
+    matchLabels:
+      app: ark
+      component: server
+  # Whether or not to snapshot volumes. This only applies to PersistentVolumes for Azure, GCE, and
+  # AWS. Valid values are true, false, and null/unset. If unset, Ark performs snapshots by default.
```
Comment (skriss) [inline docs/api-types/backup.md]: for unset, maybe say "Ark performs snapshots if and only if a persistentVolumeProvider is configured"?

File: docs/api-types/backup.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        # Array of namespaces to which this hook does not apply. Optional.
+        excludedNamespaces:
+        - some-namespace
+        # Array of resources to which this hook applies. The only resource supported at this time is
+        # pods.
+        includedResources:
+        - pods
+        # Array of resources to which this hook does not apply. Optional.
+        excludedResources: []
+        # This hook only applies to objects matching this label selector. Optional
```
Comment (skriss) [inline docs/api-types/backup.md]: `.` at the end

File: docs/api-types/backup.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            # The type of hook. This must be "exec".
+            exec:
+              # The name of the container where the command will be executed. If unspecified, the
+              # first container in the pod will be used. Optional.
+              container: my-container
+              # The command to execute, specified as an array. Required.
+              command:
+                - /bin/uname
+                - -a
+
```
Comment (skriss) [inline docs/api-types/backup.md]: remove blank line?

File: docs/api-types/backup.md (hunk 2 above)
Comment (ncdc) [reply docs/api-types/backup.md]: Reworded, ptal

File: docs/api-types/backup.md (hunk 1 above)
Comment (ncdc) [reply docs/api-types/backup.md]: Fixed

File: docs/api-types/backup.md (hunk 3 above)
Comment (ncdc) [reply docs/api-types/backup.md]: Fixed

File: docs/api-types/backup.md (hunk 4 above)
Comment (ncdc) [reply docs/api-types/backup.md]: Fixed

**Comment by skriss** [comment]
LGTM, squash -> merge.

**Comment by ncdc** [comment]
squashed

**Pull request merged**